bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/relay/RelayHandler.java

```diff
@@ -108,10 +110,26 @@ public RelayHandler(Thing thing) {
     @Override
     protected boolean processDeviceInfo(Device deviceInfo) {
         this.isInImpulseSwitchMode = isRelayInImpulseSwitchMode(deviceInfo);
-        configureChannels();
+        boolean isChannelConfigurationValid = configureChannels();
+        if (!isChannelConfigurationValid)
+            return false;
+
+        updateModePropertyIfApplicable();
         return super.processDeviceInfo(deviceInfo);
     }
 
+    private void updateModePropertyIfApplicable() {
+        String modePropertyValue = isInImpulseSwitchMode ? ImpulseSwitchService.IMPULSE_SWITCH_SERVICE_NAME
+                : PowerSwitchService.POWER_SWITCH_SERVICE_NAME;
+
+        Map<String, String> properties = getThing().getProperties();
+        if (properties.containsKey(PROPERTY_MODE) && properties.get(PROPERTY_MODE).equals(modePropertyValue)) {
+            return;
+        }
+
+        updateThing(editThing().withProperty(PROPERTY_MODE, modePropertyValue).build());
```

Comment:
Could be simplified:
```suggestion
        updateProperty(PROPERTY_MODE, modePropertyValue);
```